tuf/repository_lib.py

```diff
@@ -155,9 +155,16 @@ def _generate_and_write_metadata(rolename, metadata_filename,
     # Don't hash-prefix consistent target files if they are handled out of band
     consistent_targets = consistent_snapshot and not use_existing_fileinfo
 
-    metadata = generate_targets_metadata(targets_directory, roleinfo['paths'],
-        roleinfo['version'], roleinfo['expires'], roleinfo['delegations'],
-        consistent_targets, use_existing_fileinfo, storage_backend)
+    metadata, delegations_update = generate_targets_metadata(targets_directory,
+        roleinfo['paths'], roleinfo['version'], roleinfo['expires'],
+        roleinfo['delegations'], consistent_targets, use_existing_fileinfo,
+        storage_backend, repository_name)
+
+    # Update roledb with the latest delegations info
+    if delegations_update:
```

Comment:
Yes, the idea is to avoid calling `update_roleinfo()` for each delegation on every `write`, especially when their number can grow high if delegated hashed bins are used. 

Anyway: 
- when tested with the scrip from https://github.com/theupdateframework/tuf/issues/1005#issuecomment-601351054 the overhead when updating all delegations and not using the `delegations_update` boolean seems to be not more than ~1-2secs (tested in a regular VM on my laptop)
- assuming that delegated hashed bins all use the same key, key revocation would mean updating all delegations

So if you are hinting towards simplifying the code at the cost of some negligible efficiency, you may have a point :slightly_smiling_face: 

In general, the overall overhead of the code in this PR seems to be in the range of 1-2 seconds (too small to actually calculate an exact delta without a dedicated test setup)

Reply:
Thanks for crunching the numbers. I'd absolutely be in favour of the code simplification.